packages/wallets/src/AuthProvider.tsx

```diff
@@ -4,7 +4,29 @@ import { AuthProvider } from '@deriv/api-v2';
 const WALLETS_LOGINID_LOCALSTORAGE_KEY = 'active_wallet_loginid';
 
 const WalletsAuthProvider = ({ children, ...rest }: Omit<ComponentProps<typeof AuthProvider>, 'loginIDKey'>) => (
-    <AuthProvider {...rest} loginIDKey={WALLETS_LOGINID_LOCALSTORAGE_KEY}>
+    <AuthProvider
+        {...rest}
+        loginIDKey={WALLETS_LOGINID_LOCALSTORAGE_KEY}
+        selectDefaultAccount={accountsObject => {
+            const loginIds = Object.keys(accountsObject);
+            const result = loginIds.filter((loginId: string) => {
+                const { account_category: accountCategory, account_type: accountType } = accountsObject[loginId];
+                const isWallet = accountCategory == 'wallet';
+                const isFiat = accountType == 'doughflow';
+                return isWallet && isFiat;
+            })[0];
+
+            if (!result) {
+                const result = loginIds.filter((loginId: string) => {
```

Comment:
To avoid confusion, Maybe we can use a different name since result is already declared in outer scope